.github/workflows/pr-realm-js.yml

```diff
@@ -202,38 +219,37 @@ jobs:
-          - { os: darwin, target: test, runner: macos-latest, environment: node }
-          - { os: android, target: "test:android", runner: macos-latest, environment: react-native, arch: "armeabi-v7a" }
-          - { os: ios, target: "test:ios", runner: macos-latest, environment: react-native, arch: "apple" }
+          - {  os: linux, target: test, runner: ubuntu-latest, environment: node }
+          - {  os: linux, target: "test:main", runner: ubuntu-latest, environment: electron }
+          - {  os: linux, target: "test:renderer", runner: ubuntu-latest, environment: electron }
+          #- {  os: windows, target: test, runner: windows-latest, environment: node}
+          #- {  os: windows, target: "test:main", runner: windows-latest, environment: electron }
+          #- {  os: windows, target: "test:renderer", runner: windows-latest, environment: electron }
+          - {  os: darwin, target: "test:main", runner: macos-latest, environment: electron }
+          - {  os: darwin, target: "test:renderer", runner: macos-latest, environment: electron }
+          - {  os: darwin, target: test, runner: macos-latest, environment: node }
+          - {  os: android, target: "test:android", runner: macos-latest, environment: react-native, arch: "armeabi-v7a" }
+          - {  os: ios, target: "test:ios", runner: macos-latest, environment: react-native, arch: "apple" }
           #- { os: ios, target: "test:catalyst", runner: macos-latest, environment: react-native, arch: "apple" }
     timeout-minutes: 60
     steps:
-      - name: Generate Cluster Differentiator
-        id: differentiator
-        run: echo "::set-output name=name::${{matrix.variant.os}}-${{matrix.variant.target}}-${{matrix.variant.environment}}-${{env.GITHUB_RUN_ATTEMPT}}"
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          submodules: "recursive"
 
-      - uses: realm/ci-actions/mdb-realm/deployApps@fac1d6958f03d71de743305ce3ab27594efbe7b7
-        id: deploy-mdb-apps
+      - name: Deploy Cluster if not already deployed
+        uses: realm/ci-actions/mdb-realm/deploy@e7b65d7d587e1abc358968d7ff816cc0163906d2
+        id: deploy
         with:
-          realmUrl: ${{ secrets.REALM_QA_BASE_URL }}
           projectId: ${{ secrets.ATLAS_QA_PROJECT_ID }}
-          atlasUrl: ${{ secrets.ATLAS_QA_BASE_URL}}
           apiKey: ${{ secrets.ATLAS_QA_PUBLIC_API_KEY }}
           privateApiKey: ${{ secrets.ATLAS_QA_PRIVATE_API_KEY }}
-          differentiator: ${{ steps.differentiator.outputs.name }}
+          clusterSize: "M10"
 
-      - name: Checkout code
-        uses: actions/checkout@v2
-        with:
-          submodules: "recursive"
+      # We set it from the output from the above deployment, in case we rerun this test, which will create a new cluster name
+      - name: Set Cluster Name to Environment
+        run: echo "CLUSTER_NAME=${{ steps.deploy.outputs.clusterName }}" >> $GITHUB_ENV
```

Comment:
What will set this in case of a regular (non-retry) run?

Reply:
The above deploy step.  If the cluster has already been deployed, it will just set the output to that name.

Reply:
Ahh okay. What's the need to store it in an env? Couldn't the test step just use the output directly?

Reply:
No need 🤷🏼 Just realized it was broken for the retry case and came up with a solution with the current setup.  We can use it directly